beams/service/tree_ticker.py

```diff
@@ -208,7 +208,18 @@ def work_func(self):
                 while (self.state.get_pause_tree()):
                     time.sleep(self.state.get_tick_delay_ms())  # reusing this here.... could use a semaphore...
 
-                tick_tree(self.tree, self.state.get_tick_interactive() == TickConfiguration.INTERACTIVE, self.state.get_tick_delay_ms())
+                # If we are in interactive mode
+                if self.state.get_tick_interactive() == TickConfiguration.INTERACTIVE:
+                    # wait till the semaphore gets incremented, this is the IPC method to communicate a tick_interactive
+                    got_tick = self.tick_sem.acquire(timeout=0.2)
+                    # because of the timeout (makes cleaning up thread easier) we need to check how it timeodout
+                    if got_tick:
```

Comment:
```suggestion
                    # because of the timeout (makes cleaning up thread easier) we need to check how it timeodout
                    if self.tick_sem.acquire(timeout=0.2):
```

I'm pretty sure there's no issue putting blocking statements inside the if-condition